### ISSUE TITLE
validationResult.then is not a function

### ISSUE THREAD

**Issue by enrichit** [pqina/filepond-plugin-file-validate-type#10]
Hello I'm getting an exception on 1.2.0 and it seems like something that should affect a good portion of users so maybe I'm doing something wrong.

The issue is around https://github.com/pqina/filepond-plugin-file-validate-type/blob/master/dist/filepond-plugin-file-validate-type.js#L180.

`validationResult` is a boolean but also evaluates to true and so it doesn't return with `handleRejection()`. instead the promise gets fulfilled and then drops out of the if branch where `validationResult` is then handled as if it were a promise.

Instead I believe there should be an extra return before dropping out of the if or there should be conditional to ensure that `validationResult` is a promise at the time of execution.

This is the code that initialises my filepond fields:
```js
    import * as FilePond from 'filepond';
    import FilePondPluginImagePreview from 'filepond-plugin-image-preview';
    import FilePondPluginFileMetadata from 'filepond-plugin-file-metadata';
    import FilePondPluginFileValidateType from 'filepond-plugin-file-validate-type';

    FilePond.setOptions({
      name: 'filepond',
      server: {
        process: '/process-upload',
        load: '/load-upload/',
        revert: null
      }
    })

    FilePond.registerPlugin(
      FilePondPluginImagePreview,
      FilePondPluginFileMetadata,
      FilePondPluginFileValidateType
    );

    export default (element, type, { multiple, onFileProcessed, onFileRemoved, files, imagesOnly } = {}) => {
      const options = {
        allowMultiple: multiple === true,
        fileMetadataObject: { type },
      }

      if (files) {
        options.files = files.map(file => ({
          source: file,
          options: {
            type: 'local'
          }
        }))
      }

      if (imagesOnly == true) {
        options.acceptedFileTypes = ['image/*']
      }

      const pond = FilePond.create(element, options)

      const idMap = {}

      pond.on('processfile', (error, file) => {
        if (!error) {
          const { keys } = JSON.parse(file.serverId)
          const serverId = keys[0]

          idMap[file.id] = serverId
          onFileProcessed(serverId, file)
        }
      })

      pond.on('removefile', file => onFileRemoved(idMap[file.id], file))
    }
```
Please let me know if I can provide any additional information or a pull request. Or if this is expected behaviour and I'm doing things wrong.

**Comment by rikschennink** [comment]
Hmm, that's odd, I see what you mean, it should throw but it doesn't, it's probably silent because it's in the Promise? I'll add the return right away.

**Comment by rikschennink** [comment]
Should be fixed in `1.2.1`, please confirm.

**Comment by enrichit** [comment]
Works perfectly. It didn't really stop the plugin from working but it was throwing for me when I had pause on exceptions set which was slightly annoying.

That was a very fast response though! Thank you very much, I'll close this ticket to save you time.
